datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -178,6 +178,13 @@ impl CaseExpr {
             let when_value = self.when_then_expr[i]
                 .0
                 .evaluate_selection(batch, &remainder)?;
+            // Treat 'NULL' as false value
+            let when_value = match when_value {
+                ColumnarValue::Scalar(value) if value.is_null() => {
```

Comment:
This is a reasonable solution I think -- ideally we would have been able to coerce this value into a `ColumnarValue::Scalar(ScalarValue::Boolean(None))` at some earlier point.

However, I think this special case is reasonable (and the test coverage is nice) so 👍  from me